### ISSUE TITLE
Break apart functionality into modules

### ISSUE THREAD

**Issue by davideast** [firebase/angularfire#799]
AngularFire is currently shipped as one contained `"firebase"` module. This worked well for the `< 3.0` SDK days when auth and database were both in the `firebase.js` file. With the `3.0` SDK world brings the ability to create a build of the SDK for what features you need. AngularFire should support this, while leaving support for current users.

This will currently leave us with 3 modules:
- Auth
- Database
- Storage (future support)

**Proposal `angularfire.feature` module convention:**

```js
angular
    .module('app', ['angularfire.auth', 'angularfire.database', 'angularfire.storage'])
    .controller('MyCtrl', function MyCtrl($firebaseArray, $firebaseAuthService) {

    });
```

**Backwards compatibility**
And for existing users the `"firebase"` module can contain every existing module that AngularFire offers.

```js
angular
    .module('app', ['firebase'])
    .controller('MyCtrl', function MyCtrl($firebaseArray, $firebaseAuthService) {

    });
```

**Comment by jwngr** [comment]
+1 from me although what's your reason for using `angularfire.*` instead of `firebase.*`? We also just need to make sure the error messages are clear if and when you forget to include a certain feature.

**Comment by davideast** [comment]
The naming switch was an attempt to make the module more service specific to AngularFire, but I since we do name everything `$firebaseFeature()` we could still keep it.

**Comment by davideast** [comment]
Closed as of #841
